File: frontend/src/objects/shared/Promotion.ts
import type { PromotionDiscountType } from './PromotionDiscountType'
import type { PromotionTriggerType } from './PromotionTriggerType'

export type { PromotionDiscountType } from './PromotionDiscountType'
export type { PromotionTriggerType } from './PromotionTriggerType'

export interface Promotion {
  id: string
  title: string
  discountType: PromotionDiscountType
  discountValue: number
  triggerType: PromotionTriggerType
  triggerValue: number
  startsAt?: string | null
  endsAt?: string | null
  dailyStartTime?: string | null
  dailyEndTime?: string | null
  productIds?: string[]
  usageLimit?: number | null
  remainingUses?: number | null
  enabled: boolean
}

File: backend/src/promotion/objects/Promotion.scala
package delivery.promotion.objects

import delivery.domain.ProductId

final case class Promotion(
    id: String,
    title: String,
    discountType: PromotionDiscountType,
    discountValue: Double,
    triggerType: PromotionTriggerType,
    triggerValue: Double,
    startsAt: Option[String],
    endsAt: Option[String],
    dailyStartTime: Option[String] = None,
    dailyEndTime: Option[String] = None,
    productIds: List[ProductId] = Nil,
    usageLimit: Option[Int] = None,
    remainingUses: Option[Int] = None,
    enabled: Boolean
)
